src/libYARP_dev/src/devices/ControlBoardWrapper/RPCMessagesParser.cpp

```diff
@@ -833,10 +833,9 @@ void RPCMessagesParser::handleInteractionModeMsg(const yarp::os::Bottle& cmd,
                     delete [] modes;
                 }
                 break;
-
-                    lastRpcStamp.update();
-                    appendTimeStamp(response, lastRpcStamp);
             }
+            lastRpcStamp.update();
+            appendTimeStamp(response, lastRpcStamp);
```

Comment:
This code was never executed before, and it is being moved to a place where it will actually be executed, therefore it is potentially dangerous 😁 
@barbalberto @randaz81 can you please review this change?

Reply:
Right, but I saw the function `RPCMessagesParser::handleTorqueMsg(...)` had the same structure of this one, that's the reason why I moved these lines in the same position.

Reply:
Short answer: The change is correct, so I'll keep it.

Long aswer:  On a second look, I see that some  `handleXXX` functions are adding the timeStamp while others doesn't.
The timestamp is added as part of the message and not as an envelope, which is strange to me.

Actually in the future I'd like to move all the couples of lines
```		
  lastRpcStamp.update();		
  appendTimeStamp(response, lastRpcStamp);
```
from the `handleXXX` and put them once and for all at the end of `respond(...)` function.
And maybe use the envelope.
But it is quite a rude change... I don't know if I'm brave enough :sweat_smile: 
For now just keep this change.

Reply:
:+1: @barbalberto thanks for the reply!

Reply:
I am afraid this function is not covered by any unit test. I think it requires at least a test in the simulator, trying to switch interaction modes on the robot, or to extend one of the remote_controlboard tests to check this.

Reply:
This is adding a timestamp at the end of RPC message, in the data field (not even in the envelope).
Do you really want to add a test for it? I'd rather remove it completely, and maybe add a proper timestamp in  the envelope...